pincer/commands/commands.py

```diff
@@ -168,169 +183,337 @@ async def test_command(
+            raise InvalidArgumentAnnotation(
+                "Choice type is only allowed for str, int, and float"
             )
+        if choices is not MISSING:
+            for choice in choices:
+                if isinstance(choice.value, int) and annotation.command_type == float:
+                    continue
+                if not isinstance(choice.value, annotation.command_type):
+                    raise InvalidArgumentAnnotation(
+                        "Choice value must match the command type"
+                    )
+
+        cannel_types = annotation.get_arg(ChannelTypes)
+        if cannel_types is not MISSING and annotation.command_type != Channel:
+            raise InvalidArgumentAnnotation(
+                "ChannelTypes is only available for Channel")
 
-        if get_origin(annotation) is Choices:
-            args = get_args(annotation)
+        max_value = annotation.get_arg(MaxValue)
+        min_value = annotation.get_arg(MinValue)
 
-            if len(args) > 25:
+        for i, value in enumerate((min_value, max_value)):
+            if (
+                value is not MISSING
+                and annotation.command_type != int
+                and annotation.command_type != float
+            ):
+                t = ("MinValue", "MaxValue")
                 raise InvalidArgumentAnnotation(
-                    f"Choices/Literal annotation `{annotation}` on "
-                    f"parameter `{param}` in command `{cmd}` "
-                    f"(`{func.__name__}`) amount exceeds limit of 25 items!"
+                    f"{t[i]} is only available for int and float"
                 )
 
-            choice_type = type(args[0])
+        options.append(
+            AppCommandOption(
+                type=command_type,
+                name=param,
+                description=argument_description,
+                required=required,
+                choices=choices,
+                channel_types=cannel_types,
+                max_value=max_value,
+                min_value=min_value,
+            )
+        )
+
+    return register_command(
+        func=func,
+        app_command_type=AppCommandType.CHAT_INPUT,
+        name=name,
+        description=description,
+        enable_default=enable_default,
+        guild=guild,
+        cooldown=cooldown,
+        cooldown_scale=cooldown_scale,
+        cooldown_scope=cooldown_scope,
+        command_options=options
+    )
+
 
-            if choice_type is Descripted:
-                choice_type = type(args[0].key)
+def user_command(
+    func=None,
+    *,
+    name: Optional[str] = None,
+    enable_default: Optional[bool] = True,
+    guild: Union[Snowflake, int, str] = None,
+    cooldown: Optional[int] = 0,
+    cooldown_scale: Optional[float] = 60,
+    cooldown_scope: Optional[ThrottleScope] = ThrottleScope.USER,
+):
+    """A decorator to create a user command to register and respond to
+    with the discord API from a function.
```

Comment:
Would rather do this, unless I'm misunderstanding the use of this decorator (sorry if that's the case)
```suggestion
    """A decorator to make a user command register and respond
    with the Discord API from a function.
```